j2v/generation/generator.py

```diff
@@ -159,6 +160,22 @@ def __add_dimension(self, field_path_sql, current_view, dimension_name, dim_val,
 
         self.views_dimensions_expr[current_view].add(new_dimension)
 
-        sql_select = st.field_str_template.format(__path=field_path_sql, TABLE=current_view,
-                                                  json_type=json_type, path_alias=full_path_nice.upper())
+        if not self.handle_null_values_in_sql:
+            sql_select = st.field_str_template.format(__path=field_path_sql, TABLE=current_view,
+                                              json_type=json_type, path_alias=full_path_nice.upper())
+        if self.handle_null_values_in_sql:
+
+            if json_type == "number":
+                sql_select = st.non_nullable_numeric_field_str_template.format(__path=field_path_sql,
+                    TABLE=current_view, json_type=json_type, path_alias=full_path_nice.upper())
+
+            elif json_type == "string" and dim_type != "date_time":
+                sql_select = st.non_nullable_text_field_str_template.format(__path=field_path_sql,
+                    TABLE=current_view, json_type=json_type, path_alias=full_path_nice.upper())
+
+            else :
```

Comment:
We can change the whole if-else tree to avoid duplicating `sql_select = st.field_str_template.format(__path=field_path_sql,
                    TABLE=current_view, json_type=json_type, path_alias=full_path_nice.upper())`

Reply:
I've decided to create a private function to build and return _sql_select_

The private function has a simplified if/else tree that has less duplication and is easier to read.